slatedb/src/sst_iter.rs

```diff
@@ -314,15 +352,45 @@ impl<'a> SstIterator<'a> {
     }
 
     fn stop(&mut self) {
-        let num_blocks = self.index.borrow().block_meta().len();
-        self.next_block_idx_to_fetch = num_blocks;
+        if let Some(index) = self.index.as_ref() {
+            let num_blocks = index.borrow().block_meta().len();
+            self.next_block_idx_to_fetch = num_blocks;
+        }
         self.state.stop();
     }
+
+    async fn fetch_index(&mut self) -> Result<(), SlateDBError> {
+        if self.index.is_none() {
```

Comment:
what's the case where its expected that this would be called when index is not None?

Reply:
I don't think there is one, but it's nice to make this idempotent anyway. (e.g. `init` is designed to be idempotent and right now the call to `fetch_index` is actually outside of the `!self.state.is_initialized()` block for no good reason - just the way the code came out after a few refactors)